modules/Microsoft.App/containerApps/deploy.bicep

```diff
@@ -0,0 +1,241 @@
+param ingressAllowInsecure bool = true
+
+@description('Optional. Target Port in containers for traffic from ingress.')
+param ingressTargetPort int = 80
+
+@description('Optional. Container environment variables.')
+param environmentVar array = []
+
+@description('Required. Container App resources.')
+param containerResources object
+
+@description('Optional. Maximum number of container replicas. Defaults to 10 if not set.')
+param scaleMaxReplicas int = 1
+
+@description('Optional. Minimum number of container replicas.')
+param scaleMinReplicas int = 0
+
+@description('Optional. Scaling rules.')
+param scaleRules array = []
+
+@allowed([
+  'Multiple'
+  'Single'
+])
+@description('Optional. ActiveRevisionsMode controls how active revisions are handled for the Container app.')
+param activeRevisionsMode string = 'Single'
+
+@description('Required. Resource ID of environment.')
+param environmentId string
+
+@allowed([
+  ''
+  'CanNotDelete'
+  'ReadOnly'
+])
+@description('Optional. Specify the type of lock.')
+param lock string = ''
+
+@description('Optional. Tags of the resource.')
+param tags object = {}
+
+@description('Optional. Collection of private container registry credentials for containers used by the Container app.')
+param registries array = []
+
+@description('Optional. Enables system assigned managed identity on the resource.')
+param systemAssignedIdentity bool = false
+
+@description('Optional. The set of user assigned identities associated with the resource, the userAssignedIdentities dictionary keys will be ARM resource ids and The dictionary values can be empty objects ({}) in requests.')
+param userAssignedIdentities object = {}
+
+@description('Optional. Array of role assignment objects that contain the \'roleDefinitionIdOrName\' and \'principalId\' to define RBAC role assignments on this resource. In the roleDefinitionIdOrName attribute.')
+param roleAssignments array = []
+
+@description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
+param enableDefaultTelemetry bool = false
+
+var identityType = systemAssignedIdentity ? (!empty(userAssignedIdentities) ? 'SystemAssigned,UserAssigned' : 'SystemAssigned') : (!empty(userAssignedIdentities) ? 'UserAssigned' : 'None')
+
+var identity = identityType != 'None' ? {
+  type: identityType
+  userAssignedIdentities: !empty(userAssignedIdentities) ? userAssignedIdentities : null
+} : null
+
+resource defaultTelemetry 'Microsoft.Resources/deployments@2021-04-01' = if (enableDefaultTelemetry) {
+  name: 'pid-47ed15a6-730a-4827-bcb4-0fd963ffbd82-${uniqueString(deployment().name, location)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+    }
+  }
+}
+
+@description('Optional. Custom domain bindings for Container App hostnames.')
```

Comment:
These parameters should be before the variables and telemetry resource

Reply:
moved